log4j2/src/main/java/com/google/common/flogger/backend/log4j2/Log4j2MetadataKeyValueHandlers.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2019 The Flogger Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ package com.google.common.flogger.backend.log4j2;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.flogger.MetadataKey;
+import com.google.common.flogger.backend.MetadataHandler;
+
+import java.util.Iterator;
+
+/** 
+ * A helper class providing the default callbacks and handlers for processing metadata as key/value
+ * pairs in order to use them with Log4j2.
+ */
+final class Log4j2MetadataKeyValueHandlers {
+    private static final MetadataHandler.ValueHandler<Object, Log4j2KeyValueHandler> EMIT_METADATA =
```

Comment:
Yes, I used language level 1.6 because of the BUILD settings. For log4j2/BUILD I updated them to use jdk 1.8.  I was not sure what the minimum supported jdk version is, but I also did not ask.